src/results.cpp

```diff
@@ -228,6 +228,43 @@ util::Optional<T> Results::try_get(size_t ndx)
     return util::none;
 }
 
+Results::IteratorWrapper::IteratorWrapper(IteratorWrapper const& rgt)
+{
+    *this = rgt;
+}
+
+Results::IteratorWrapper& Results::IteratorWrapper::operator=(IteratorWrapper const& rgt)
+{
+    m_ndx = rgt.m_ndx;
+    if (rgt.m_it)
+        m_it = std::make_unique<Table::ConstIterator>(*rgt.m_it);
+    return *this;
+}
+
+Obj Results::IteratorWrapper::get(Table const& table, size_t ndx)
+{
+    // Using a Table iterator is much faster for sequential access into a table
+    // than indexing into it, but aren't random access and there's nontrivial
+    // overhead to creating one. Experimentally you need to read at least 3
+    // rows for an iterator to be faster, it's very unlikely that trying to use
+    // an iterator for random-access cases will be worthwhile, and it stops
+    // being faster to increment a pre-existing iterator if it's more than
+    // 50-100 rows away.
+    // As such we only create iterators when starting from index 0 and only
+    // advance pre-existing iterators if they're within 50 rows of the
+    // requested index.
+    if (ndx == 0 && (!m_it == m_ndx != 0) && table.size() > 2) {
```

Comment:
I have a hard time getting my head around `(!m_it == m_ndx != 0)`. What does it mean in practice? Initially `!m_it` is true and `m_ndx != 0` is false, so when can the expression become true?
When running the unit tests, we never get an iterator created.

Reply:
I appear to have "simplified" this expression at the last moment and actually just turned it into nonsense instead.

There aren't any tests for Results::get() because originally objectstore was tested exclusively via SDK tests, and even after adding testing to OS we only wrote tests for newly added features and not the things ported from existing SDKs. Good time to change that, though.